### PR TITLE
fix(FrameworkElement): [macOS] Ensure we set parent to a DependencyObject

### PR DESCRIPTION
**Pull request by kazo0**
closes https://github.com/unoplatform/Uno.Gallery/issues/129
closes https://github.com/unoplatform/Uno.Gallery/issues/130

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?

Bindings are not propagated properly when using a ScrollViewer since the FrameworkElement Implementation of `ViewDidMoveToWindow` tries to set the 

## What is the new behavior?

Bindings are properly propagated 

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/IFrameworkElementImplementation.macOS.tt [hunk 1]
```diff
@@ -69,17 +70,18 @@ namespace <#= mixin.NamespaceName #>
...
 					{
-						if(_superViewRef?.GetTarget() == null && superView != null)
+						var parent = GetDependencyObjectAncestor(Superview) as NSView;
+						parent ??= Superview;
+						if(_superViewRef?.GetTarget() == null && parent != null)
 						{
-							_superViewRef = new WeakReference<NSView>(superView);
-							SyncBinder(superView, newWindow);
-							((IDependencyObjectStoreProvider)this).Store.Parent = superView;
+							_superViewRef = new WeakReference<NSView>(parent);
```
Comment (kazo0) [inline src/Uno.UI/UI/Xaml/IFrameworkElementImplementation.macOS.tt:82]: Still not sure about what should be set here, the "real" Superview or the ancestor that we may have found that is a `DependencyObject`?

Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/IFrameworkElementImplementation.macOS.tt:82]: I'm not sure either - it only seems to be checked here, and TBH I'm not clear on what the intent of the check is.

Any thoughts @jeromelaban ?

Comment (kazo0) [reply src/Uno.UI/UI/Xaml/IFrameworkElementImplementation.macOS.tt:82]: But even for the `_superViewRef`? We shouldn't set that to the actual Superview regardless of it being a DependencyObject?

Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/IFrameworkElementImplementation.macOS.tt:81]: @kazo0 I would leave it as you have it, setting it to `parent` (the actual dependency object), but I'm still not sure what the check is even for.

**Review by davidjohnoliver — APPROVED** [review]

**Pull request merged**